components/core/src/Grep.cpp

```diff
@@ -603,8 +609,13 @@ std::optional<Query> Grep::process_raw_query(
         return std::nullopt;
     }
 
-    query.set_sub_queries(std::move(sub_queries));
-    return query;
+    return Query{
+            search_begin_ts,
+            search_end_ts,
+            ignore_case,
+            processed_search_string,
+            std::move(sub_queries)
+    };
```

Comment:
Is the convention here to explicitly state the type when using {} initializer for returns?

Reply:
Good point, although the method returns a `std::optional` so we need to be explicit about the type.